Base.rte/Activities/BunkerBreach.lua

```diff
@@ -2,213 +2,263 @@
+						guard = self:CreateInfantry(techID, loadoutName);
+					end
+					if guard then
+						guard.Pos = defenderBox.Center;
+						guard.Team = self.defenderTeam;
+						guard.AIMode = Actor.AIMODE_SENTRY;
+						if loadoutName == "Engineer" then
+							guard.AIMode = Actor.AIMODE_GOLDDIG;
+						end
+						MovableMan:AddActor(guard);
+					end
 				end
-			else
-				break;
 			end
 		end
 	end
+	for actor in MovableMan.AddedActors do
+		if actor.Team ~= self.defenderTeam and not actor:IsInGroup("Brains") then
+			MovableMan:ChangeActorTeam(actor, self.defenderTeam);
+		end
+	end
+end
+
+function BunkerBreach:SetupDefenderInternalReinforcementAreas()
+	if self.AI.isDefenderTeam then
+		local internalReinforcementsArea = SceneMan.Scene:GetOptionalArea("Internal Reinforcements");
+		if internalReinforcementsArea ~= nil then
+			self.AI.internalReinforcementsDoorParticle = CreateMOSRotating("Animated Background Door", "Base.rte");
+			self.AI.internalReinforcementPositions = {};
+			for internalReinforcementsBox in internalReinforcementsArea.Boxes do
+				local backgroundDoor = CreateTerrainObject("Module Back Middle E", "Base.rte");
+				backgroundDoor.Pos = SceneMan:SnapPosition(internalReinforcementsBox.Corner, true);
+				self.AI.internalReinforcementPositions[#self.AI.internalReinforcementPositions + 1] = backgroundDoor.Pos + Vector(24, 24);
+				SceneMan:AddSceneObject(backgroundDoor);
+			end
+		end
+	end
+end
 
+function BunkerBreach:SetupFogOfWar()
 	if self:GetFogOfWarEnabled() then
 		SceneMan:MakeAllUnseen(Vector(24, 24), self.attackerTeam);
 		SceneMan:MakeAllUnseen(Vector(24, 24), self.defenderTeam);
-		if self.CPUTeam ~= -1 then
-			SceneMan:MakeAllUnseen(Vector(70, 70), self.CPUTeam);
-			--Assume that the AI has scouted the terrain
-			for x = 0, SceneMan.SceneWidth - 1, 65 do
-				SceneMan:CastSeeRay(self.CPUTeam, Vector(x, 0), Vector(0, SceneMan.SceneHeight), Vector(), 1, 9);
-			end
+		
+		for x = 0, SceneMan.SceneWidth - 1, 24 do
+			SceneMan:CastSeeRay(self.attackerTeam, Vector(x, 0), Vector(0, SceneMan.SceneHeight), Vector(), 1, 9);
 		end
-		--Lift the fog around friendly actors
+		
+		for mainBunkerBox in self.mainBunkerArea.Boxes do
+			SceneMan:RevealUnseenBox(mainBunkerBox.Corner.X, mainBunkerBox.Corner.Y, mainBunkerBox.Width, mainBunkerBox.Height, self.defenderTeam);
+		end
+
 		for actor in MovableMan.AddedActors do
-			for ang = 0, math.pi * 2, 0.1 do
-				SceneMan:CastSeeRay(actor.Team, actor.EyePos, Vector(130 + FrameMan.PlayerScreenWidth * 0.5, 0):RadRotate(ang), Vector(), 1, 4);
+			for angle = 0, math.pi * 2, 0.05 do
+				SceneMan:CastSeeRay(actor.Team, actor.EyePos, Vector(150 + FrameMan.PlayerScreenWidth * 0.5, 0):RadRotate(angle), Vector(), 1, 4);
 			end
 		end
 	end
 end
 
+function BunkerBreach:StartActivity()
+	collectgarbage("collect");
+
+	self.attackerTeam = Activity.TEAM_1;
+	self.defenderTeam = Activity.TEAM_2;
+
+	-- Because of game oddities, we need to set funds to match starting gold manually.
```

Comment:
Oof. Any chance we can fix this in cpp?

Reply:
It's an artifact of the Activity and GameActivity having 2 different ways to set money.
It can be fixed of course, but the options are to make breaking changes to other activities that rely on activity funds setting for specific teams, or to do more work and have both options be safe.

I didn't wanna make breaking changes, and I didn't wanna scope creep more and add more stuff that needs to be reviewed for what is ultimately low priority, so I'm gonna leave it be and add an issue to be dealt with in future.

Reply:
Sounds good to me! No complaints about that.